rust/src/ffiutil.rs

```diff
@@ -109,6 +109,24 @@ pub(crate) fn ffi_ptr_array_to_vec<'a, T>(a: *mut glib_sys::GPtrArray) -> Vec<*m
     v
 }
 
+/// Transform a NULL-terminated array of C strings to an allocated Vec holding unowned references.
+/// This is similar to FromGlibPtrContainer::from_glib_none(), but avoids cloning elements.
+pub(crate) fn ffi_strv_to_os_str_vec<'a>(mut strv: *mut *mut libc::c_char) -> Vec<&'a OsStr> {
```

Comment:
Hm I am increasingly feeling that we should explicitly mark any function like this that defeats lifetimes as `unsafe`.  But it's not new here so OK as is.